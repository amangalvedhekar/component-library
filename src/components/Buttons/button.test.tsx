import React from 'react';
import { shallow, mount } from 'enzyme';
import SomeButton from "./Button";

describe("Button component", () => {
    it("should render", () => {
        const wrapper = shallow(<SomeButton>Hello Button</SomeButton>);
        expect(wrapper).toMatchSnapshot();
    });
    it("should call click", () => {
        const handleClick = jest.fn();
        const wrapper = mount(<SomeButton onClick={handleClick}>Hello Button</SomeButton>);
        wrapper.find("button").simulate("click");
        expect(handleClick).toBeCalled();
    });
});
