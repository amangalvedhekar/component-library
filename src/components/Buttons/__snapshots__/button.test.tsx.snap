// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  background: ",
          [Function],
          ";
  color: ",
          [Function],
          ";
  cursor: pointer;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid blue;
  border-radius: 3px;
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Hello Button
</StyledComponent>
`;
